kratos/mpi/tests/sources/test_mpi_assign_unique_model_part_collection_tag_utility.cpp

```diff
@@ -0,0 +1,113 @@
+            ModelPart& r_sub_modelpart_4 = r_model_part.CreateSubModelPart("YSubModelPart4");
+
+            r_model_part.AddNodalSolutionStepVariable(PARTITION_INDEX);
+            CppTestsUtilities::Create2DGeometry(r_model_part, "Element2D3N");
+
+            r_sub_modelpart_1.AddNode(r_model_part.pGetNode(1));
+            r_sub_modelpart_2.AddNode(r_model_part.pGetNode(4));
+            r_sub_modelpart_3.AddNode(r_model_part.pGetNode(5));
+            r_sub_modelpart_4.AddNode(r_model_part.pGetNode(6));
+
+            // Adding nodes to random submodelparts
+            if (DataCommunicator::GetDefault().IsDistributed()) {
+                if (rank == 0) {
+                    r_sub_modelpart_3.AddNode(r_model_part.pGetNode(4));
+                }
+                else {
+                    r_sub_modelpart_3.AddNode(r_model_part.pGetNode(6));
+                    r_sub_modelpart_2.AddNode(r_model_part.pGetNode(5));
+                    r_sub_modelpart_4.AddNode(r_model_part.pGetNode(5));
+
+                }
+            } else {
+                r_sub_modelpart_3.AddNode(r_model_part.pGetNode(4));
+                r_sub_modelpart_3.AddNode(r_model_part.pGetNode(6));
+                r_sub_modelpart_2.AddNode(r_model_part.pGetNode(5));
+                r_sub_modelpart_4.AddNode(r_model_part.pGetNode(5));
+            }
+
+            // Adding elements to random submodelparts
+            r_sub_modelpart_1.AddElement(r_model_part.pGetElement(1));
+            r_sub_modelpart_2.AddElement(r_model_part.pGetElement(4));
+
+
+            AssignUniqueModelPartCollectionTagUtility collections_utility(r_model_part);
+
+            IndexIndexMapType nodes_tags, conds_tags, elems_tags;
+            IndexStringMapType collections;
+            collections_utility.ComputeTags(nodes_tags, conds_tags, elems_tags, collections);
+
+            const std::string filename = "mpi_test";
+
+            AssignUniqueModelPartCollectionTagUtility::WriteTagsToJson(filename + std::to_string(rank), collections);
+            KRATOS_CHECK_EQUAL(collections.size(), 8);
+
+            DataCommunicator::GetDefault().Barrier();
+            for (int i = 0; i < size; i++) {
+                if (i != rank) {
+                    IndexStringMapType read_collections;
+                    AssignUniqueModelPartCollectionTagUtility::ReadTagsFromJson(filename + std::to_string(i), read_collections);
+
+                    KRATOS_CHECK_EQUAL(collections.size(),read_collections.size());
+
+                    for (IndexType j = 0; j < read_collections.size(); j++) {
+                        KRATOS_CHECK_EQUAL(collections[j], read_collections[j]);
+                    }
+                }
+            }
+            DataCommunicator::GetDefault().Barrier();
+            Kratos::filesystem::remove(FilesystemExtensions::JoinPaths({FilesystemExtensions::CurrentWorkingDirectory(), filename + std::to_string(rank) + ".json"}));
```

Comment:
this should crash if called from multiple ranks 
=> call only from rank 0

Reply:
Note that each rank is deleting its own file. Why should it crash?

 I can remove all on rank 0 if it is preferable

Reply:
ah I didn't get that, sorry for the noise